aten/src/THC/THCIntegerDivider.cuh

```diff
@@ -2,6 +2,9 @@
 #define THC_INTEGER_DIVIDER_INC
 
 #include <assert.h>
+#if defined(__CUDA_ARCH__) || defined(__HIP_DEVICE_COMPILE__)
```

Comment:
I think we don't need `__HIP_DEVICE_COMPILE__`, the hipify script will turn `__CUDA_ARCH__` to `__HIP_ARCH__` which will be defined in rocm build

Reply:
I see the substitution for [`__CUDACC__`](https://github.com/pytorch/pytorch/blob/master/tools/amd_build/pyHIPIFY/cuda_to_hip_mappings.py#L282) but not `__CUDA_ARCH__`.

Here's the hipified header: [THHIntegerDivider.cuh](https://gist.github.com/colesbury/5868ec820057e0027f0f79cd97e226ad)